### ISSUE TITLE
installing fgwas

### ISSUE THREAD

**Issue by billgreenwald** [frazer-lab/cluster#122]
I am trying to install fgwas.  The first step is to run the configure script inside the folder (which is in my software/fgwas-0.3.6 directory).  I can't get past the error:

checking for boostlib >= 1.42... configure: We could not detect the boost libraries (version 1.42 or higher). If you have a staged boost library (still not installed) please specify $BOOST_ROOT in your environment and do not give a PATH to --with-boost option.  If you are sure you have boost installed, then check your version number looking in <boost/version.hpp>. See http://randspringer.de/boost for more documentation.

The webpage doesn't load anything for me, and i've tried using --with-boost=[my path to boost]

the paths ive tried for boost are 
/software
/software/boost
/software/boost/libs
/software/boost/stage
/software/boost/stage/lib

The boost installer finished and output the following, if that helps:

The following directory should be added to compiler include paths:

```
/frazer01/home/bill/software/boost
```

The following directory should be added to linker library paths:

```
/frazer01/home/bill/software/boost/stage/lib
```

**Comment by tatarsky** [comment]
From what I see you describe above I would suggest trying:

```
export BOOST_ROOT=/frazer01/home/bill/software/boost
```

That assumes you have a lib/include tree in that location. You would have to make sure that is correct. 

I will however look at your configure script perhaps later today if that doesn't work.

Your other paths BTW are definitely not right as there is no such thing as `/software/boost`. We have no shared boost module. I repeat from an earlier git I am happy to build one.

**Comment by billgreenwald** [comment]
I have tried setting BOOST_ROOT to all of the paths i listed above as well, with no avail

**Comment by tatarsky** [comment]
I'd have to look at it then or one of the local sysadmins. Won't be for a bit.

**Comment by billgreenwald** [comment]
Ok.  Just to clarify, the paths that I tried were all in reference to /frazer01/home/bill

so the paths I tried from the first post were really:

/frazer01/home/bill/software
/frazer01/home/bill/software/boost
/frazer01/home/bill/software/boost/libs
/frazer01/home/bill/software/boost/stage
/frazer01/home/bill/software/boost/stage/lib

**Comment by billgreenwald** [comment]
The github for fgwas also recommends trying to run with LDFLAGS=-L/[path]

I tried with all the above paths, as well as /frazer01/home/bill/software/boost/boost

still did not work

**Comment by cdeboever3** [comment]
If @tatarsky is going to spend time making this work, I think making common
boost and fgwas modules makes sense.

**Comment by tatarsky** [comment]
Thats fine with me. Please point me to fgwas source or I can google it. I am not familiar with the package. Still won't be for a bit. Backlogged.

**Comment by billgreenwald** [comment]
https://github.com/joepickrell/fgwas

**Comment by tatarsky** [comment]
OK. So one item fell out of the way this afternoon so I'm going to play by play this in hopes it helps. I am happy to review what you did to see the difference but we'll have to arrange a time to share a tty. 
1. Downloaded the released tar ball of fgwas 0.3.6
2. Noted with configure it wants more recent boost. Fairly typical. Same basic statement on system RPMS and such items being best separated. 
3. Downloaded the boost 1.60.0 tar ball

I know from the past this requires newer gcc. So 

```
module add gcc
./bootstrap.sh --prefix=/frazer01/software/boost-1.60.0
./b2
(go compose lengthy poetry about birds and trees, decide its not good enough, discard)
./b2 install 
```

This leaves you with a boost tree in a "traditional" lib/include format which is very important for autoconf as autoconf won't try real hard to find stuff. 

```
/frazer01/software/boost-1.60.0
  include lib
```
1. Build the fgwas package

```
./configure --with-boost=/frazer01/software/boost-1.60.0 --prefix=/frazer01/software/fgwas-0.3.6
make
make install
/frazer01/software/fgwas-0.3.6/bin/fgwas 

fgwas v. 0.3.6
by Joe Pickrell (jkpickrell@nygenome.org)
```

Module in a moment.

**Comment by tatarsky** [comment]
BTW the above assumes by executing "it works". I have real idea how to test it but can look or feel free. Also note I need to add the GSL libraries to puppet. 

```
module add fgwas
```

**Comment by tatarsky** [comment]
I stand corrected. They already were there. But did add the -devel versions to the head nodes for building items GSL needing.

**Comment by billgreenwald** [comment]
I never ran ./b2 install as from the instructions I was following there was no call to install--that was probably the problem.

Thanks for the help!

**Comment by tatarsky** [comment]
Boost is very odd in its build/install methodologies. Its very non-GNU autoconf.

I've saved those steps based on dealing with this same dependency issue in a few places. I don't recall where I originally derived them. 

I am reviewing if I should make a couple of developer modules that set the commonly used autoconf CPPFLAGS and LDFLAGS or not. 

Some configure packages that allow a custom boost have an explicit `--with-boost=` however that usually "works" better.

Basically boost is weird. Not the first time I've seen it cause such matters.

**Comment by tatarsky** [comment]
Assuming this is good to go. Closing. Re-open if not the case.
